plutus-benchmark/validation/Main.hs

```diff
@@ -1,89 +1,120 @@
-mkBgroup :: String -> [(Int, (Int, Int, Int, Int))] -> Benchmark
-mkBgroup dirname bms = bgroup dirname (map (mkBM dirname) bms)
 
+{-- | Benchmarks based on validations obtained using
+plutus-use-cases:plutus-use-cases-scripts, which runs various contracts on the
+blockchain simulator and dumps the applied validators as flat-encoded
+scripts. Generating these scripts is a very lengthy process involving building a
+lot of code, so the scripts were generated once and copied to the 'data'
+directory here.
+
+NB. Running these benchmarks with `stack bench` will use copies of the scripts
+in `.stack_work` (and accessed via Paths_plutus_benchmark), and if a file in
+`data` is removed and the benchmarks are re-run, the benchmarking code may still
+be able to access the old copy in stack's files.  --}
+
+{- | The name of the directory where the scripts are kept.  This must match the
+   location of the files relative to the directory containing the cabal file.
+   IF THE DIRECTORY IS MOVED, THIS MUST BE UPDATED. -}
+getScriptDirectory :: IO FilePath
+getScriptDirectory = do
+  root <- getDataDir
+  return $ root </> "validation" </> "data"
+
+{- | Subdirectories for different contracts. The hierarchical arrangement makes
+   it easier to group benchmarks. Any changes here should be reflected in the
+   "data-files" section of the cabal file to ensure that Paths_plutus_benchmark
+   still works. -}
+contractDirs :: [FilePath]
+contractDirs =
+    [ "crowdfunding"
+    , "currency"
+    , "escrow"
+    , "future"
+    , "game-sm"
+    , "multisig-sm"
+    , "ping-pong"
+    , "prism"
+    , "pubkey"
+    , "stablecoin"
+    , "token-account"
+    , "vesting"
+    , "marlowe" </> "trustfund"  -- The Marlowe examples aren't included in plutus-use-cases-scripts
+    , "marlowe" </> "zerocoupon"
+    ]
+
+-- | A small subset of the contracts for quick benchmarking
+contractDirs2 :: [FilePath]
+contractDirs2 =
+    [ "crowdfunding"
+    , "prism"
+    , "token-account"
+    , "marlowe" </> "zerocoupon"
+    ]
+
+
+type Term          = UPLC.Term    PLC.Name      PLC.DefaultUni PLC.DefaultFun ()
+type Program       = UPLC.Program PLC.Name      PLC.DefaultUni PLC.DefaultFun ()
+type DbProgram     = UPLC.Program UPLC.DeBruijn PLC.DefaultUni PLC.DefaultFun ()
+
+fromDeBruijn :: DbProgram -> IO Program
+fromDeBruijn prog = do
+    let namedProgram = UPLC.programMapNames (\(UPLC.DeBruijn ix) -> UPLC.NamedDeBruijn "v" ix) prog
+    case PLC.runQuote $ runExceptT @UPLC.FreeVariableError $ UPLC.unDeBruijnProgram namedProgram of
+      Left e  -> errorWithoutStackTrace $ show e
+      Right p -> return p
+
+loadFlat :: FilePath -> IO Term
+loadFlat file = do
+  contents <- BSL.fromStrict <$> BS.readFile file
+  case unflat contents of
+    Left e  -> errorWithoutStackTrace $ "Flat deserialisation failure for " ++ file ++ ": " ++ show e
+    Right r -> do
+        p <- fromDeBruijn r
+        return . force $  UPLC.toTerm p
```

Comment:
this only forces when `force` is evaluated. I think you might want
```suggestion
        return $! force $  UPLC.toTerm p
```
to be safe